### PR TITLE
{phys}[GCC/11.3.0] Geant4 v11.1.2

### PR DESCRIPTION
**Pull request by akesandgren**
(created using `eb --new-pr`)

This superseedes #9656 among others
It also correctly (I hope) handles the Geant4 data files and env vars in the geant4.sh and geant4make.sh scripts

This should also take care of the issue in  https://github.com/easybuilders/easybuild-easyblocks/issues/1352

### CODE REVIEW

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.10
See https://gist.github.com/akesandgren/3796d29d6c688f333f254636401976a9 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
bwd-rockylinux-92 - Linux Rocky Linux 9.2 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), 2 x NVIDIA NVIDIA GeForce GTX 1060 6GB, 525.105.17, Python 3.9.16
See https://gist.github.com/SebastianAchilles/e75869189f6e9bb86292351abc45c821 for a full test report.

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
himem06.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Platinum 8160 CPU @ 2.10GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/jfgrimm/0c23820dcd200e670b5d22c900143883 for a full test report.

**Comment by jfgrimm** [comment]
main difference I see to the version our site installs is that we also include dependencies on `Xerces-C++`, `Qt5`, and build with additional configopts `-DGEANT4_USE_GDML=ON -DXERCESC_ROOT_DIR=$XERCESC_ROOT_DIR -DGEANT4_USE_NETWORKVRML=ON -DGEANT4_USE_OPENGL_X11=ON -DGEANT4_BUILD_MULTITHREADED=ON -DGEANT4_USE_RAYTRACER_X11=ON -DGEANT4_USE_QT=ON`

**Comment by akesandgren** [comment]
@jfgrimm NETWORKVRML doesn't exist any longer according to a grep -r, besides that this new version should be good I think

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.10
See https://gist.github.com/akesandgren/b8c708a4f96aaec81c7e5c8782c6bf40 for a full test report.

File: easybuild/easyconfigs/g/Geant4-data/Geant4-data-11.1.eb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+local_datasets = [
+    ('G4NDL', '4.7', 'G4NDL', 'G4NEUTRONHPDATA'),  # NDL
+    ('G4EMLOW', '8.2', 'G4EMLOW', 'G4LEDATA'),  # Low energy electromagnetics
+    ('PhotonEvaporation', '5.7', 'G4PhotonEvaporation', 'G4LEVELGAMMADATA'),  # Photon evaporation
+    ('RadioactiveDecay', '5.6', 'G4RadioactiveDecay', 'G4RADIOACTIVEDATA'),  # Radioisotopes
+    ('G4SAIDDATA', '2.0', 'G4SAIDDATA', 'G4SAIDXSDATA'),  # SAID
+    ('G4PARTICLEXS', '4.0', 'G4PARTICLEXS', 'G4PARTICLEXSDATA'),  # Particle XS - replaces Neutron XS
+    ('G4PII', '1.3', 'G4PII', 'G4PIIDATA'),  # PII
+    ('RealSurface', '2.2', 'G4RealSurface', 'G4REALSURFACEDATA'),  # Optical Surfaces
+    ('G4SAIDDATA', '2.0', 'G4SAIDDATA', 'G4SAIDXSDATA'),  # SAID
```
Comment (smoors) [inline easybuild/easyconfigs/g/Geant4-data/Geant4-data-11.1.eb:21]: `G4SAIDDATA` is listed twice, is that intentional?

Comment (akesandgren) [reply easybuild/easyconfigs/g/Geant4-data/Geant4-data-11.1.eb:21]: (name-of-dataset, version, name-part-of-topdir-of-unpacked-tarfile, envvarname)

So it's not listed twice.

Comment (akesandgren) [reply easybuild/easyconfigs/g/Geant4-data/Geant4-data-11.1.eb:21]: Oh... that I didn't notice... fixing.

Comment (akesandgren) [reply easybuild/easyconfigs/g/Geant4-data/Geant4-data-11.1.eb]: @smoors @jfgrimm better now?

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
@boegelbot please test @ jsc-zen2

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @akesandgren!

**Pull request merged**